tests/test_repository.py

```diff
@@ -190,39 +190,52 @@ async def update(self, dto: AccessRequest) -> None:
         self.last_upsert = dto
 
 
-class NotificationRecord(NamedTuple):
-    """Class that records a sent notification while testing."""
+class EventPublisherDummy(EventPublisherPort):
+    """Dummy event publisher for testing."""
 
-    recipient: str
-    subject: str
-    text: str
-
-
-class NotificationEmitterDummy(NotificationEmitterPort):
-    """Dummy notification emitter for testing."""
-
-    notifications: dict[str, NotificationRecord]
+    events: dict[AccessRequestDetails, str]
 
     def reset(self) -> None:
-        """Reset the recorded notification."""
-        self.notifications = {}
+        """Reset the recorded events."""
+        self.events = {}
```

Comment:
Maybe better to use a list? Then you would also catch if the same events are sent out twice.

Reply:
Changes are in [6ba34d3](https://github.com/ghga-de/access-request-service/pull/14/commits/6ba34d329a02de775cd90731239135b05e8bed41), good idea.

Reply:
Actually I meant just a plain list, not a (default)dict of lists.

But if you prefer the defaultdict, then you don't need to catch KeyErrors.

Reply:
Please see [6ba34d3](https://github.com/ghga-de/access-request-service/pull/14/commits/6ba34d329a02de775cd90731239135b05e8bed41), where I switched to a list and store a more direct mock event